crypto/digest_extra/digest_test.cc

```diff
@@ -527,44 +528,176 @@ TEST(DigestTest, TransformBlocks) {
     uint8_t hash2[HASH_NAME##_DIGEST_LENGTH];                                 \
     EXPECT_TRUE(HASH_NAME##_Final(hash2, &ctx3)) << "final 3";                \
                                                                               \
+    /* finally check the resulting hash matches the baseline hash */          \
     EXPECT_EQ(Bytes(hash1), Bytes(hash2)) << "same hash";                     \
   }
 
-// The macro DIGEST_TEST_InitAndGetState_Body only contains the body of the
-// tests and not the TEST(...) { part, so that IDE can still easily recognize
-// those as tests.
 
-TEST(DigestTest, InitAndGetStateMD5) {
-  DIGEST_TEST_InitAndGetState_Body(MD5, MD5_CTX, MD5_CBLOCK);
+// DIGEST_TEST_InitAndGetStateLarge_Body expands to the body of the various
+// InitAndGetStateLarge* tests below.
+// Compared to DIGEST_TEST_InitAndGetStateBasic_Body, it allows testing
+// hashing an arbitrary amount of data.
+// Of particular interest is the choice of NB_1000_BLOCKS to be such that
+// the number of hash bits n > 2^32, therefore overflowing uint32_t (used for
```

Comment:
I understand that you're trying to overflow `Nl` so that `Nh` gets updated and the hash continues operation seamlessly and successfully. If that's the case, I think whenever overflow is mentioned, here and further below, it may need to be clarified that it's not an error condition, just the normal operation.